src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -90,36 +73,60 @@ public void removeMetric(KafkaMetric metric) {
         metrics.remove(metric.metricName());
     }
 
-    String metricName(MetricName metricName) {
-        String prefix = this.prefix
-                .replace('.', '_')
-                .replace('-', '_')
-                .toLowerCase(Locale.ROOT);
-        String group = metricName.group()
-                .replace('.', '_')
-                .replace('-', '_')
-                .toLowerCase(Locale.ROOT);
-        String name = metricName.name()
-                .replace('.', '_')
-                .replace('-', '_')
+    /**
+     * Called when the Prometheus server scrapes metrics.
+     * @return metrics that match the configured allow list
+     */
+    @Override
+    public MetricSnapshots collect() {
+        Map<String, GaugeSnapshot.Builder> gaugeBuilders = new HashMap<>();
+        Map<String, InfoSnapshot.Builder> infoBuilders = new HashMap<>();
+
+        for (Map.Entry<MetricName, KafkaMetric> entry : metrics.entrySet()) {
+            MetricName metricName = entry.getKey();
+            KafkaMetric kafkaMetric = entry.getValue();
+            LOG.trace("Collecting Kafka metric {}", metricName);
+
+            String prometheusMetricName = metricName(metricName);
+            // TODO Filtering should take labels into account
+            if (!config.isAllowed(prometheusMetricName)) {
+                LOG.info("Kafka metric {} is not allowed", prometheusMetricName);
+                continue;
+            }
+            LOG.info("Kafka metric {} is allowed", prometheusMetricName);
+            Labels labels = labelsFromTags(metricName.tags());
+
+            Object valueObj = kafkaMetric.metricValue();
+            if (valueObj instanceof Number) {
+                double value = ((Number) valueObj).doubleValue();
+                GaugeSnapshot.Builder builder = gaugeBuilders.computeIfAbsent(prometheusMetricName, k -> GaugeSnapshot.builder().name(prometheusMetricName));
+                builder.dataPoint(DataPointSnapshotBuilder.convert(value, labels));
+            } else {
+                InfoSnapshot.Builder builder = infoBuilders.computeIfAbsent(prometheusMetricName, k -> InfoSnapshot.builder().name(prometheusMetricName));
+                builder.dataPoint(DataPointSnapshotBuilder.convert(valueObj, labels, metricName.name()));
+            }
+        }
+        List<MetricSnapshot> snapshots = new ArrayList<>();
+        for (GaugeSnapshot.Builder builder : gaugeBuilders.values()) {
+            snapshots.add(builder.build());
+        }
+        for (InfoSnapshot.Builder builder : infoBuilders.values()) {
+            snapshots.add(builder.build());
+        }
+        return new MetricSnapshots(snapshots);
+    }
+
+    private String metricName(MetricName metricName) {
+        return PrometheusNaming
```

Comment:
I'm not sure I fully understand how to use the various methods in `PrometheusNaming` especially `sanitizeMetricName()` and `prometheusName()`.